### ISSUE TITLE
Find alternative to keytar package

### ISSUE THREAD

**Issue by laurent22** [laurent22/joplin#8829]
The [keytar package](https://github.com/atom/node-keytar) we use for keychain access is deprecated and they don't mention an alternative.

We should look for one, either a fork or a new better package.

Over time that discontinued package might be an issue since things might break from one OS release to the next. We have for example this crash: https://github.com/laurent22/joplin/issues/8819

**Comment by personalizedrefrigerator** [comment]
It looks like [vscode used to use keytar and now uses the electron safeStorage API](https://code.visualstudio.com/docs/editor/settings-sync#_troubleshooting-keychain-issues).

**Comment by laurent22** [comment]
Perfect! If they use this it's safe for us to use it too. Although it means we may have to keep keytar for CLI, since it won't have access to Electron's APIs.

**Comment by Daeraxa** [comment]
For what it is worth the other project I've been involved with in the last year or so is Pulsar which has, for better or worse, also inherited these other repos like Keytar as part of the Atom sunset. 

We still rely on Keytar for our `github` package but so far haven't needed to make any changes so we still use the original GitHub published npm package. However we have spoken about the situation and we have provisionally agreed that if we needed to make any improvements to Keytar (and anything else in a similar situation like https://github.com/pulsar-edit/keyboard-layout)  we wouldn't only make those changes available in the https://github.com/pulsar-edit/node-keytar package but also publish it to npm for anybody who wanted to migrate from the original and take advantage of any improvements.

Basically what I'm saying is that the Pulsar org will likely try maintain our fork of Keytar but probably only in "maintenance-only" mode rather than pushing active development. Currently issues aren't enabled on the repo but I'll bring this conversation up, in the meantime you can post on the forum https://github.com/orgs/pulsar-edit/discussions if there is anything critical (or even if not).

**Comment by RandellBrianKnight** [comment]
A possibility to explore is Passport, a simple, unobtrusive authentication for Node.js. It may be found at https://github.com/jaredhanson/passport
